### ISSUE TITLE
Strange warning around atoms when checking a concatenated string

### ISSUE THREAD

**Issue by noozo** [rrrene/credo#884]
When running credo against the following piece of code:

```
{:error, %Invoices.Error{code: :invalid_prorate_to, details: details}} ->
        {:error,
         %Error{
           code: :invoice_period_inconsistent,
           details: %{
             invoice: %{
               start_day: details.invoice_start_day,
               end_day: details.end_day
             },
             shorten_to_day: shorten_to_day
           },
           message:
             "The billing period of the booking's latest invoice does not match the " <>
               "Stripe subscription's current billing period."
         }}
```

Credo will complain with the following warning:

```
warning: Unexpected error while parsing lib/<my_project>/<offending_module>.ex: [%Credo.Issue{category: :error, check: Credo.Code.ParserError, column: nil, diff_marker: nil, exit_status: 0, filename: "nofilename", line_no: [line: 289, column: 62], message: "atom length must be less than system limit: ", meta: [], priority: 0, scope: nil, severity: nil, trigger: nil}]
  (credo 1.5.6) lib/credo/check/readability/string_sigils.ex:53: Credo.Check.Readability.StringSigils.run/2
  (credo 1.5.6) lib/credo/check/readability/string_sigils.ex:5: Credo.Check.Readability.StringSigils.do_run_on_source_file/3
  (elixir 1.12.1) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
  (elixir 1.12.1) lib/task/supervised.ex:35: Task.Supervised.reply/5
  (stdlib 3.15.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
```

Using credo 1.5.6.

**Comment by rrrene** [comment]
I cannot reproduce this by adding this file to a project:

```elixir
defmodule Invoices.Error do
  defstruct code: nil, details: nil, message: nil

  alias Invoices.Error

  def foo(bar, shorten_to_day) do
    case bar do
      {:error, %Invoices.Error{code: :invalid_prorate_to, details: details}} ->
        {:error,
         %Error{
           code: :invoice_period_inconsistent,
           details: %{
             invoice: %{
               start_day: details.invoice_start_day,
               end_day: details.end_day
             },
             shorten_to_day: shorten_to_day
           },
           message:
             "The billing period of the booking's latest invoice does not match the " <>
               "Stripe subscription's current billing period."
         }}
    end
  end
end
```

What's interesting in your stacktrace is the `message` that says:

>  "atom length must be less than system limit: "

Does this ring any bells?

**Comment by rrrene** [comment]
Closing due to inactivity. Please re-open at your convenience :+1:
